### PR TITLE
Fix error sink msg duplication & false positives

### PR DESCRIPTION
**Pull request by Daniel-VDM**
## Issue

Currently, the error sink just dumps any error to a ring buffer and does not remove any error report from the buffer if the tx ended up being successful. Some downstream tools (like CLI and block explorer) use the sink to check if a recent transaction failed and short circuits a failure msg if it does not. Therefore, it is important that the error sink does not duplicate error msgs and removes error messages for txs that made it through.

Moreover, I exposed the TxPool stats RPC.

SS of RPC:

Screen Shot 2020-05-05 at 3 19 46 AM

## Test

It is sufficient to test this on localnet as you can cause errors as needed. This was done & a unit test was added.

### Unit Test Coverage

Before:

```
PASS
coverage: 26.5% of statements
ok  	github.com/harmony-one/harmony/core	7.030s
```

After:

```
PASS
coverage: 26.2% of statements
ok  	github.com/harmony-one/harmony/core	6.991s
```

## Operational Checklist

1. **Does this PR introduce backward-incompatible changes to the on-disk data structure and/or the over-the-wire protocol?**. (If no, skip to question 8.)

    **NO**

8. **Does this PR introduce backward-incompatible changes *NOT* related to on-disk data structure and/or over-the-wire protocol?** (If no, continue to question 11.)

    **NO**

11. **Does this PR introduce significant changes to the operational requirements of the node software, such as >20% increase in CPU, memory, and/or disk usage?**

    **NO**

### CODE REVIEW

File: internal/configs/sharding/testnet.go [hunk 1]
```diff
@@ -82,5 +82,5 @@ var testnetReshardingEpoch = []*big.Int{
 	params.TestnetChainConfig.StakingEpoch,
 }
 
-var testnetV0 = MustNewInstance(4, 25, 25, numeric.OneDec(), genesis.TNHarmonyAccounts, genesis.TNFoundationalAccounts, testnetReshardingEpoch, TestnetSchedule.BlocksPerEpoch())
+var testnetV0 = MustNewInstance(4, 30, 25, numeric.OneDec(), genesis.TNHarmonyAccounts, genesis.TNFoundationalAccounts, testnetReshardingEpoch, TestnetSchedule.BlocksPerEpoch())
```
Comment (rlan35) [inline internal/configs/sharding/testnet.go:85]: why is this changed?

Comment (gupadhyaya) [inline internal/configs/sharding/testnet.go:85]: why is this change?

Comment (Daniel-VDM) [reply internal/configs/sharding/testnet.go:85]: It's to fix all the unit tests that use the test blockchain. We need to have slots for BLS keys so that one can send staking txs in the prestaking epoch. (previously we would get not sufficient slots for BLS keys -- hence the old TODO msgs)

**Comment by janet-harmony** [comment]
Jenkins, test this please
